bliss/surveys/dc2.py

```diff
@@ -1,55 +1,217 @@
+import collections.abc  # noqa: WPS301
 import math
+import multiprocessing
 import pathlib
 import random
 
 import numpy as np
 import pandas as pd
 import torch
 from astropy.io import fits
+from astropy.io.fits import Header
 from astropy.wcs import WCS
 from einops import rearrange
-from torch.utils.data import DataLoader
+from torch.utils.data import DataLoader, Dataset
 
 from bliss.catalog import FullCatalog
 from bliss.surveys.survey import Survey
 
 
+def from_wcs_header_str_to_wcs(wcs_header_str: str):
+    return WCS(Header.fromstring(wcs_header_str))
+
+
+def map_nested_dicts(cur_dict, func):
+    if isinstance(cur_dict, collections.abc.Mapping):
+        return {k: map_nested_dicts(v, func) for k, v in cur_dict.items()}
+
+    return func(cur_dict)
+
+
+def generate_split_file(image_index, object_attributes):
```

Comment:
Is there a more descriptive variable name than `object_attributes` we could use here? (Or could we break up the argument into multiple arguments?)

Reply:
change it to `self_copy` in commit a521a09